app/services/organizations/importers/csv_import_service.rb

```diff
@@ -4,61 +4,66 @@ module Organizations
   module Importers
     class CsvImportService
       Status = Data.define(:success?, :count, :no_match, :errors)
-      def initialize(file)
+      def initialize(file, organization)
         @file = file
-        @organization = Current.organization
+        @organization = organization
         @count = 0
         @no_match = []
         @errors = []
       end
 
       def call
         catch(:halt_import) do
-          validate_file
-
-          CSV.foreach(@file.to_path, headers: true, skip_blanks: true).with_index(2) do |row, index|
-            # Header may be different depending on which form applicaiton was used(e.g. google forms) or how it was created(User creates form with "Email Address")
-            email = row[@email_header].downcase
-            # Google forms uses "Timestamp", other services may use a different header
-            csv_timestamp = Time.parse(row["Timestamp"]) if row["Timestamp"].present?
-
-            person = Person.find_by(email:, organization: @organization)
-            previously_matched_form_submission = FormSubmission.where(person:, csv_timestamp:)
-
-            if person.nil?
-              @no_match << [index, email]
-            elsif previously_matched_form_submission.present?
-              next
-            else
-              ActiveRecord::Base.transaction do
-                create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
-                @count += 1
+          @file.download do |data|
```

Comment:
Background jobs should use a file that has been uploaded to a storage service. This steams the download and yields the chunks.